### ISSUE TITLE
Go to definition doesn't find the implementation until the file is opened once, even though I'm using a single compile_commands.json

### ISSUE THREAD

**Issue by tamirzb** [clangd/clangd#846]
Essentially, it looks like my issue is exactly like #802. The thing is that in that case it seems that the problem was having multiple `compile_commands.json` files. In my case, I only have a single one but the issue still exists.
This happens after clangd finished indexing everything and is idle.

Basically what happens is:
1. I go to definition for a function, clangd finds its declaration in a header file.
2. Even when I'm in the header file, trying to go to definition for that functions still directs me to the same declaration in the header file.
3. Only after I open the file with the implementation of the function manually, then going to definition really directs me to the implementation of that function (and each time I do it it toggles between the header and the implementation, which is the expected behavior).

**System information**

clangd version 12.0.1 & neovim 0.5 with built-in LSP

**Comment by HighCommander4** [comment]
Does the source file containing the definition have an entry in the `compile_commands.json`?

**Comment by tamirzb** [comment]
Ok so after looking at this more, I figured out this is due to symlinks in the path, so basically a duplicate of #503.
It doesn't look like the patches there are going to be accepted, so I guess I'll either have to build my own fork/find a workaround :/

**Comment by MrKleeblatt** [comment]
Hello there.
I have the exact same problem but there is no symlink in my project structure.
I use a `.clangd` file instead of `compile_commands.json` which contains only these few lines
```yaml
CompileFlags:
  Compiler: clang
  Add:
  - -g
  # - --std=gnu2x
  - -Wall
  - -Wextra
  - -Wpedantic
  - -Werror
```

`clangd version 18.1.8`

Is there any progress on this issue?

**Comment by HighCommander4** [comment]
A `compile_commands.json` is required for clangd to index the project, and cross translation unit features (like finding a definition in a different translation unit) require an index. This is by design, and there are no plans to change this.

**Comment by HighCommander4** [comment]
To clarify: only files listed in `compile_commands.json` will be indexed. It's where clangd gets the list of files to index.

**Comment by MrKleeblatt** [comment]
@HighCommander4 Thank you! Seems like I missed that…
